aws/logs_monitoring/enhanced_lambda_metrics.py

```diff
@@ -108,25 +128,30 @@ def _refresh(self):
             )
             return
 
-        self.tags_by_arn = build_tags_by_arn_cache()
-        self.missing_arns -= set(self.tags_by_arn.keys())
+        tags_fetched, last_modified = get_cache_from_s3()
 
-    def _is_expired(self):
-        """Returns bool for whether the tag fetch TTL has expired"""
-        earliest_time_to_refetch_tags = (
-            self.last_tags_fetch_time + self.tags_ttl_seconds
-        )
-        return time() > earliest_time_to_refetch_tags
+        # s3 cache fetch succeeded
+        if last_modified > -1:
+            self.tags_by_arn = tags_fetched
 
-    def _should_refresh_if_missing_arn(self, resource_arn):
-        """Determines whether to try and fetch a missing lambda arn.
-        We only refresh if we encounter an arn that we haven't seen
-        since the last refresh. This prevents refreshing on every call when
-        tags can't be found for an arn.
-        """
-        if resource_arn in self.missing_arns:
-            return False
-        return self.tags_by_arn.get(resource_arn) is None
+        if self._is_expired(last_modified):
+            send_forwarder_internal_metrics("s3_cache_expired")
```

Comment:
Can we also add a debug log wherever you emit the telemetry? Because the telemetry does get aggregated and lose some precision and information.

Reply:
✅